File: Easy/2714-left-and-right-sum-differences/left-and-right-sum-differences.java
class Solution{
    public int[] leftRightDifference(int[] nums){
        int n=nums.length;

        int[] leftsum=new int[n];
        Arrays.fill(leftsum, 0);
        int[] rightsum=new int[n];
        Arrays.fill(rightsum, 0);

        for(int i=1; i<n; i++){
            leftsum[i]=leftsum[i-1]+nums[i-1];
        }
        for(int i=n-2; i>=0; i--){
            rightsum[i]=rightsum[i+1]+nums[i+1];
        }

        int[] res=new int[n];
        for(int i=0; i<n; i++){
            // System.out.println(leftsum[i] + " " + rightsum[i]);
            res[i]=Math.abs(leftsum[i]-rightsum[i]);
        }

        return res;

    }
}

File: 2714-left-and-right-sum-differences/left-and-right-sum-differences.py
class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        n=len(nums)

        leftsum=[0]*n
        rightsum=[0]*n
        for i in range(1, n):
            leftsum[i]=leftsum[i-1]+nums[i-1]
        for i in range(n-2, -1, -1):
            rightsum[i]=rightsum[i+1]+nums[i+1]

        res=[0]*n
        for i in range(n):
            res[i]=abs(leftsum[i]-rightsum[i])

        return res